app/assets/javascripts/admin/addon/components/highlighted-code.gjs

```diff
@@ -0,0 +1,26 @@
+import Component from "@glimmer/component";
+import { service } from "@ember/service";
+import { modifier } from "ember-modifier";
+import highlightSyntax from "discourse/lib/highlight-syntax";
+
+export default class HighlightedCode extends Component {
+  @service session;
+  @service siteSettings;
+
+  highlight = modifier(async (element) => {
+    const code = document.createElement("code");
+    code.classList.add(`lang-${this.args.lang}`);
+    code.textContent = this.args.code;
+
+    const pre = document.createElement("pre");
+    pre.appendChild(code);
+
+    await highlightSyntax(pre, this.siteSettings, this.session);
+
+    element.replaceChildren(pre);
```

Comment:
In the previous implementation, the raw unhighlighted code would be shown while highlightjs is being loaded. Could we replicate that here? Maybe good enough to add the element to the DOM before running highlightjs?

```suggestion
    element.replaceChildren(pre);
    await highlightSyntax(pre, this.siteSettings, this.session);
```